tasks/ast_tools/src/derives/clone_in.rs

```diff
@@ -60,52 +62,92 @@ fn derive_enum(def: &EnumDef) -> TokenStream {
         }
     };
 
-    impl_clone_in(&ty_ident, def.has_lifetime, &alloc_ident, &body)
+    impl_clone_in(&ty_ident, def.has_lifetime, &alloc_ident, &body, &quote!())
```

Comment:
You will need a `clone_in_with_semantic_ids` impl for most enums too. If you call `clone_in_with_semantic_ids` on a `&Program` you want it to be using `clone_in_with_semantic_ids` all the way through the AST, not diverting to `clone_in` as soon as it hits a `Statement`.

Otherwise `Program::clone_in_with_semantic_ids` will do exactly the same as `Program::clone_in` - cloned AST won't have any IDs.